feat: add search on docs

tedim52 opened the pull request:
## Description
- enables users to search through our docs
- with search, users should be able to navigate/find info in our docs much more easily

## User Facing
- [x] Yes

tedim52 committed:
add algolia config

tedim52 committed:
remove comments

tedim52 committed:
Merge branch 'main' into tedi/docsearch

h4ck3rk3y commented on docs/docusaurus.config.js:
```diff
@@ -156,6 +156,27 @@ const config = {
+      algolia: {
+        appId: 'NTSX40VZB8',
+
+        // Public API key, safe to commit
```
useful comment! :)

leeederek commented:
Hey @tedim52, this is great. Search functionality on our docs is much needed! Two questions:
1. Is the theme of the search pop-up box something we can change? It looks like it might be specific to Algolia, but just wondering because the theme style doesn't exactly match the rest of the docs and website.
2. When testing functionality using [branch preview URL](https://tedi-docsearch.kurtosis-docs.pages.dev), I wasn't able to test out the workflow/experience of this search functionality. Do you know of a way I can test this before merging?
<img width="553" alt="image" src="https://user-images.githubusercontent.com/103802618/225024020-c211435f-557e-4725-949c-a9dca52b94ba.png">

h4ck3rk3y commented on docs/docusaurus.config.js:
```diff
@@ -156,6 +156,27 @@ const config = {
+
+        contextualSearch: true,
+
+        externalUrlRegex: 'external\\.com|domain\\.com',
```
what does this dO?

tedim52 replied:
`// Optional: Specify domains where the navigation should occur through window.location instead on history.push. Useful when our Algolia config crawls multiple documentation sites and we want to navigate with window.location.href to them.`

tedim52 replied:
ended up removing cause I don't know if the options needed, can add back later if needed.

h4ck3rk3y commented on docs/docusaurus.config.js:
```diff
@@ -156,6 +156,27 @@ const config = {
+
+        externalUrlRegex: 'external\\.com|domain\\.com',
+
+        replaceSearchResultPathname: {
```
what does this do?

tedim52 replied:
`// Optional: Replace parts of the item URLs from Algolia. Useful when using the same search index for multiple deployments using a different baseUrl. You can use regexp or string in the `from` param. For example: localhost:3000 vs myCompany.com/docs` - ended up removing cause I don't know if the options needed, can add back later if needed.

h4ck3rk3y reviewed (commented):
Seems like search is tied into version so I got no result when I tried the page. @tedim52 lets see if we can index more often; have shared fears on Slack

Failure mode - Gyani releases new docs on Wednesday Morning
Angelo at Meta tries using Docs on Friday with Kurtosis but it isn't working, he can't find Starlark; he assumes no Starlark docs
On Tuesday things get indexed

tedim52 committed:
add gh workflow to reindex website on release

tedim52 committed:
remove unused params

tedim52 committed:
Merge branch 'main' into tedi/docsearch

tedim52 committed:
align search bar style with website

tedim52 commented:
@h4ck3rk3y - addressed your comment by adding a gh workflow that reindexes website everytime version docs are published.

@leeederek - addressed your comment by making the search box background, the same dark black color as website.

tedim52 committed:
reindex website only after pushing docs

tedim52 committed:
change reindex step to job

tedim52 committed:
fix styling on search

tedim52 committed:
Merge branch 'main' into tedi/docsearch

h4ck3rk3y commented on docs/src/css/custom.css:
```diff
@@ -52,4 +52,45 @@ main {
+          var(--ifm-color-emphasis-200) 0%,
+          var(--ifm-color-emphasis-100) 100%
+  );
 }
```
new line please

h4ck3rk3y commented on .github/workflows/push-docs.yml:
```diff
@@ -33,3 +33,17 @@ jobs:
+          method: 'POST'
+          username: ${{ secrets.ALGOLIA_CRAWLER_USER_ID }}
+          password: ${{ secrets.ALGOLIA_CRAWLER_API_KEY }}
+          customHeaders: '{"Content-Type": "application/json"}'
```
chatted on slack; we are going for reindex-main and reindex-yml

leeederek reviewed (approved):
Looks solid, thanks!

tedim52 committed:
separate push docs into two yamls

tedim52 committed:
Merge branch 'tedi/docsearch-2' into tedi/docsearch

tedim52 committed:
update worfklow name

h4ck3rk3y commented on .github/workflows/reindex-website-main.yml:
```diff
@@ -0,0 +1,19 @@
+jobs:
+  reindex-website-main:
+    # Only reindex if a new version release is occurring, initiated by the release PR
+    if: contains(github.head_ref, 'release-please')
```
this doesn't exist on PRs

h4ck3rk3y replied:
https://docs.github.com/en/actions/learn-github-actions/contexts

h4ck3rk3y replied:
dont think you need this at all

h4ck3rk3y commented on .github/workflows/reindex-website-main.yml:
```diff
@@ -0,0 +1,19 @@
+      - 'main'
+jobs:
+  reindex-website-main:
+    # Only reindex if a new version release is occurring, initiated by the release PR
```
off comment

h4ck3rk3y replied:
the PR looks green; maybe its within comment and GH is being funny

tedim52 replied:
removed anyways haha

tedim52 committed:
add new lines

h4ck3rk3y commented on .github/workflows/reindex-website-release.yml:
```diff
@@ -0,0 +1,19 @@
+jobs:
+  reindex-website-release:
+    # Only reindex if a new version release is occurring, initiated by the release PR
+    if: contains(github.head_ref, 'release-please')
```
don't think you need this at all

tedim52 committed:
change head_ref to ref

tedim52 committed:
remove release please check

tedim52 committed:
Merge branch 'main' into tedi/docsearch
